patches/minecraft/net/minecraft/data/AdvancementProvider.java.patch

```diff
@@ -0,0 +1,38 @@
+--- a/net/minecraft/data/AdvancementProvider.java
++++ b/net/minecraft/data/AdvancementProvider.java
+@@ -24,11 +24,15 @@
+    private static final Gson field_204024_b = (new GsonBuilder()).setPrettyPrinting().create();
+    private final DataGenerator field_204025_c;
+    private final List<Consumer<Consumer<Advancement>>> field_204283_d = ImmutableList.of(new EndAdvancements(), new HusbandryAdvancements(), new AdventureAdvancements(), new NetherAdvancements(), new StoryAdvancements());
++   private net.minecraftforge.common.data.ExistingFileHelper fileHelper;
+ 
++   @Deprecated
+    public AdvancementProvider(DataGenerator p_i48869_1_) {
+       this.field_204025_c = p_i48869_1_;
+    }
+ 
++   public AdvancementProvider(DataGenerator generatorIn, net.minecraftforge.common.data.ExistingFileHelper fileHelperIn){ this.field_204025_c = generatorIn; this.fileHelper = fileHelperIn;}
++
+    public void func_200398_a(DirectoryCache p_200398_1_) throws IOException {
+       Path path = this.field_204025_c.func_200391_b();
+       Set<ResourceLocation> set = Sets.newHashSet();
+@@ -47,12 +51,19 @@
+          }
+       };
+ 
++      registerAdvancement(consumer, fileHelper);
++
++      if(this.getClass() == AdvancementProvider.class)//Forge: Subclasses don't need this.
+       for(Consumer<Consumer<Advancement>> consumer1 : this.field_204283_d) {
+          consumer1.accept(consumer);
+       }
+ 
+    }
+ 
++   protected void registerAdvancement(Consumer<Advancement> consumer, net.minecraftforge.common.data.ExistingFileHelper fileHelper){
++
```

Comment:
Is this supposed to be empty?

Reply:
I understand this is for subclasses to override, but seems like you could at least put a javadoc or something here explaining that, as someone else reading this may not understand it.

Reply:
yes it should be empty since all vanilla Advancements are in a list and don't need to be generated again since they would be generated in the loop in the method above. Also I can't make the function abstract. 
I will add a javadoc to it any specifics i should add?

Reply:
Better option is to do what I did with the old version of the recipe provider.
And just move that for loop into this `accept(Consumer<Advancmenet>)` function.
So it's obvious that you should override the function if you want to not have the vanilla stuff.
This also fixes the annoying getClass() test you have to noop the vanilla stuff.